src/connection_check.py

```diff
@@ -35,24 +35,36 @@ def zookeeper_connected(charm: CharmBase) -> bool:
     return True
 
 
+def broker_active(unit: Unit, zookeeper_config: Dict[str, str]) -> bool:
+    """Checks ZooKeeper for client connections, checks for specific broker id.
+
+    Args:
+        unit: the `Unit` to check connection of
+        data: the relation data provided by ZooKeeper
+
+    Returns:
+        True if broker id is recognised as active by ZooKeeper. Otherwise False.
+    """
+    broker_id = unit.name.split("/")[1]
+    return broker_active_internal(broker_id, zookeeper_config)
+
 @retry(
     # retry to give ZK time to update its broker zNodes before failing
     wait=wait_fixed(5),
-    stop=stop_after_attempt(3),
+    stop=stop_after_attempt(20),
     retry_error_callback=(lambda state: state.outcome.result()),
     retry=retry_if_not_result(lambda result: True if result else False),
 )
-def broker_active(unit: Unit, zookeeper_config: Dict[str, str]) -> bool:
+def broker_active_internal(broker_id: str, zookeeper_config: Dict[str, str]) -> bool:
     """Checks ZooKeeper for client connections, checks for specific broker id.
 
```

Comment:
What's the motivation for having this be an internal method, over just public?

Reply:
I want to retain the broker_active_ prefix. Any suggestions on naming this?

Reply:
Fixed